src/lib/storage/storage_manager.cpp

```diff
@@ -260,4 +267,295 @@ std::ostream& operator<<(std::ostream& stream, const StorageManager& storage_man
+
+  for (auto header_index = size_t{0}; header_index < file_header.chunk_count; ++header_index) {
+    file_header.chunk_ids[header_index] = map[fixed_mapped_element_count + header_index];
+    file_header.chunk_offset_ends[header_index] = map[fixed_mapped_element_count + StorageManager::CHUNK_COUNT + header_index];
+  }
+
+  return file_header;
+}
+
+chunk_header StorageManager::read_chunk_header(const std::string filename, const uint32_t segment_count, const uint32_t chunk_offset_begin) {
+  // TODO: Remove need to map the whole file.
+  chunk_header header;
+  const auto map_index = element_index(chunk_offset_begin, 4);
+
+  auto fd = int32_t{};
+  Assert((fd = open(filename.c_str(), O_RDONLY)) >= 0, "Opening of file failed.");
+
+  const auto file_bytes = std::filesystem::file_size(filename);
+  auto* map = reinterpret_cast<uint32_t*>(mmap(NULL, file_bytes, PROT_READ, MAP_PRIVATE, fd, off_t{0}));
+  Assert((map != MAP_FAILED), "Mapping of Chunk Failed.");
+  close(fd);
+
+  header.row_count = map[map_index];
+
+  for (auto header_index = size_t{1}; header_index < segment_count + 1; ++header_index) {
+    header.segment_offset_ends.emplace_back(map[header_index + map_index]);
+  }
+
+  return header;
+}
+
+std::shared_ptr<Chunk> StorageManager::map_chunk_from_disk(const uint32_t chunk_offset_end, const std::string filename, const uint32_t segment_count) {
+  auto segments = pmr_vector<std::shared_ptr<AbstractSegment>>{};
+
+  auto fd = int32_t{};
+  Assert((fd = open(filename.c_str(), O_RDONLY)) >= 0, "Opening of file failed.");
+
+  const auto file_bytes = std::filesystem::file_size(filename);
+
+  // TODO: Remove unneccesary map on whole file
+  auto* map = reinterpret_cast<uint32_t*>(mmap(NULL, file_bytes, PROT_READ, MAP_PRIVATE, fd, off_t{0}));
+  Assert((map != MAP_FAILED), "Mapping of File Failed.");
+  close(fd);
+
+  const auto header = read_chunk_header(filename, segment_count, chunk_offset_end);
+  const auto header_offset = element_index(chunk_offset_end, 4);
+
+  for (auto segment_index = size_t{0}; segment_index < segment_count; ++segment_index) {
+    auto segment_offset_end = CHUNK_HEADER_BYTES(segment_count);
+    if (segment_index > 0) {
+      segment_offset_end = header.segment_offset_ends[segment_index - 1];
+    }
+
+    const auto segment_element_offset_index = element_index(segment_offset_end, 4);
+    const auto dictionary_size = map[header_offset + segment_element_offset_index];
+    const auto attribute_vector_size = map[header_offset + segment_element_offset_index + 1];
+
+    auto dictionary_values = pmr_vector<int32_t>(dictionary_size);
+    memcpy(dictionary_values.data(), &map[header_offset + segment_element_offset_index + 3], dictionary_size * sizeof(uint32_t));
+    auto dictionary = std::make_shared<pmr_vector<int32_t>>(dictionary_values);
+
+    const auto encoding_type = map[header_offset + segment_offset_end / 4 + 2];
+    (void) encoding_type;
+
+    switch (encoding_type) {
```

Comment:
Nice!
This will happen inside the DictSegment in the future.

Reply:
But we are still missing out on the 32 Bit, aren't we?